socorro/processor/processor_app.py

```diff
@@ -282,8 +263,15 @@ def _transform(self, crash_id):
             # so we can't reference `sys.exc_info()` later.
             exc_type, exc_value, exc_tb = sys.exc_info()
 
-            # Capture the error
-            self._capture_error(crash_id, exc_type, exc_value, exc_tb)
+            # PolyStorage can throw a PolyStorageException which is a sequence
+            # of exc_info items.
+            if isinstance(exc_value, collections.Sequence):
+                exc_info = exc_value
+            else:
+                exc_info = [(exc_type, exc_value, exc_tb)]
+
+            for exc_info_item in exc_info:
+                self._capture_error(crash_id, exc_info_item)
```

Comment:
So we're doing `_capture_error` with both `sys.exc_info()` and all of its items individually?

Reply:
The short answer is "yes".

The long answer is that this error handling is super muddled because (I claim) the interface here isn't well defined. Running `_transform()` could raise anything. In some cases, that exception could implement the sequence protocol and have a bunch of internal exceptions. The original code captured all of those and this version does, too.

Should it do that? I don't think so. I think we shouldn't have a notion of an exception that's a sequence of exceptions--that's weird. `PolyCrashStorage` should be handling the errors as they happen. That's a bigger change, though, and I'd need to skulk around the crashstorage code to figure out whether that's ok or not. Plus it'd probably involve rewriting a lot of tests.

Reply:
I see, thanks.